### PR TITLE
(hotfix/#34): Clear the output when the error is corrected.

### LINKED ISSUE #34: Get rid of Pylint Output Info when Error is Corrected
screenshot from 2018-10-27 17-56-57

### PR DESCRIPTION
**Pull request by feat7**
Fixes #34 
Also shows 'Write code to see something!' message when the code is not written.

### CODE REVIEW

File: static/js/javascript.js [hunk 1]
```diff
@@ -1,143 +1,198 @@
...
-        $(id).click(function (e) {
-            editor.setValue(text);
-            editor.focus(); // so that F5 works, hmm
-        });
-    };
+    function print_result(data) {
+      if (/\S/.test(editorValue)) {
+        document.getElementById("output").innerHTML = "";
+        $("#output").append("<pre>" + data + "</pre>");
+      } else document.getElementById("output").innerHTML = "Write code to see something!";
```
Comment (ethanchewy) [inline static/js/javascript.js]: So, the issue occurs during the `check` function is called not when the `run` function is called. 
 
Your logic makes sense but it shouldn't be put in the print_result but rather this check should be put in `check_syntax` specifically to check if the Editor is empty or if there are no errors in outputted by the function

Thanks for the pull request and let me know if you have any questions!

**Comment by ethanchewy** [comment]
Thanks for the pull request

Left a comment. Please let me know if you have any questions!

The main issue is that this check should be put during the AJAX `check_code` call not during the `run_code` and should check if the Editor is empty or if the error list is empty

**Comment by feat7** [comment]
The real problem is not anywhere in the java script code. It is in the python code to run the code.
When the input is empty, the stdin or stdout gives the previous input that is available there. 
ie. It doesn't flush previous stdin/stdout.

**Comment by ethanchewy** [comment]
Error fixed: https://github.com/ethanchewy/PythonBuddy/commit/43bd71a2db78757385f072e34b0660ddba0e71f3

**Pull request closed without merging**